### PR TITLE
[R2] Add aws-sdk-go-v2 example

### PR DESCRIPTION
**Pull request by KianNH**
An example demonstrating how to connect to R2's S3 API using `aws-sdk-go-v2`.

~~NOTE: This example was made to work using a patched `aws-sdk-go-v2` pending a bug fix in R2 that won't require this workaround, so it should not be merged yet.~~ The bug in R2 has been fixed in production so this PR can be safely merged.

Since it's a bit of a longer example and I'm not massively experienced with Go (or the code style of it), please suggest any improvements/formatting/etc.

cc: @ImJasonH @vlovich

### CODE REVIEW

File: content/r2/examples/aws-sdk-go.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"log"
+)
+
+func main() {
+	var BucketName = "sdk-example"
+	var AccountId = "<accountid>"
+	var AccessKeyId = "<access_key_id>"
+	var AccessKeySecret = "<access_key_secret>"
+
+	R2Resolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
```
Comment (imjasonh) [inline content/r2/examples/aws-sdk-go.md]: FWIW I was able to get this to work with just:

```
config.WithEndpointResolverWithOptions(
  s3.EndpointResolverFromURL(fmt.Sprintf("https://%s.r2.cloudflarestorage.com", accountID)),
```

without having to change the `SigningRegion`.

File: content/r2/examples/aws-sdk-go.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"log"
+)
+
+func main() {
+	var BucketName = "sdk-example"
```
Comment (imjasonh) [inline content/r2/examples/aws-sdk-go.md]: Go style nit: normally these would be lowercase, and with initialisms all uppercase, e.g., `accountID`, `accessKeySecret`, etc.

File: content/r2/examples/aws-sdk-go.md (hunk 1 above)
Comment (imjasonh) [inline content/r2/examples/aws-sdk-go.md]: Same here, if you keep `R2Resolver` I'd rename it `r2Resolver`.

File: content/r2/examples/aws-sdk-go.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		Bucket: aws.String(BucketName),
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, object := range ListObjectsOutput.Contents {
+		obj, _ := json.MarshalIndent(object, "", "\t")
+		fmt.Println(string(obj))
+	}
```
Comment (imjasonh) [inline content/r2/examples/aws-sdk-go.md]:
```suggestion
	listObjectsOutput, err := client.ListObjectsV2(context.TODO(), &s3.ListObjectsV2Input{
		Bucket: &bucketName,
	})
	if err != nil {
		log.Fatal(err)
	}

	for _, object := range listObjectsOutput.Contents {
		obj, _ := json.MarshalIndent(object, "", "\t")
		fmt.Println(string(obj))
	}
```

File: content/r2/examples/aws-sdk-go.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+	ListBucketsOutput, err := client.ListBuckets(context.TODO(), &s3.ListBucketsInput{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, object := range ListBucketsOutput.Buckets {
+		obj, _ := json.MarshalIndent(object, "", "\t")
+		fmt.Println(string(obj))
+	}
```
Comment (imjasonh) [inline content/r2/examples/aws-sdk-go.md]:
```suggestion
	listBucketsOutput, err := client.ListBuckets(context.TODO(), &s3.ListBucketsInput{})
	if err != nil {
		log.Fatal(err)
	}

	for _, object := range listBucketsOutput.Buckets {
		obj, _ := json.MarshalIndent(object, "", "\t")
		fmt.Println(string(obj))
	}
```

**Review by imjasonh — COMMENTED** [review]
Very nice!

Should we wait for the `Accept-Encoding` issue to be resolved before publishing this? Or do you want to include the middleware hacks until then?

**Comment by KianNH** [comment]
Funnily enough, I had all of the variables `likeThat` but the `SigningRegion` threw me off so I thought they should all be `LikeThis`.

File: content/r2/examples/aws-sdk-go.md (hunk 1 above)
Comment (KianNH) [reply content/r2/examples/aws-sdk-go.md]: Huh, interesting - without it in the past, the `STRING TO SIGN` just had nothing so it ended up like this.

image

Will try without.

Comment (KianNH) [reply content/r2/examples/aws-sdk-go.md]: Works without :+1:

**Review by imjasonh — APPROVED** [review]

**Review by vlovich — APPROVED** [review]
Seems fine to me after a cursory pass. If there's issues I'd prefer to just fix them up later.

**Comment by vlovich** [comment]
What accept-encoding issue?

**Comment by KianNH** [comment]
Was relating to the issue where FL replaced `Accept-Encoding: identity` with `gzip` so the signature was wrong.

**Review by deadlypants1973 — APPROVED** [review]

**Pull request merged**